File: cdo_tournament_backend/src/main/java/com/example/cdo_tournament_backend/service/JugadorPartidoService.java
package com.example.cdo_tournament_backend.service;

import java.util.List;

import com.example.cdo_tournament_backend.dto.JugadorPartidoDTO;

public interface JugadorPartidoService {

    List<JugadorPartidoDTO> getAllJugadoresPartido();
    JugadorPartidoDTO createJugadorPartido(JugadorPartidoDTO jugadorPartidoDTO);
    JugadorPartidoDTO getJugadorPartidoById(int id);
    JugadorPartidoDTO updateJugadorPartido(int id, JugadorPartidoDTO jugadorPartidoDTO);
    void deleteJugadorPartido(int id);
  
}

File: cdo_tournament_frontend/src/services/JugadorPartidoService.js
import axios from 'axios';

const JUGADORPARTIDO_API_BASE_URL = "http://localhost:8080/jugadorPartido";

class JugadorPartidoService {

    getJugadoresPartido() {
        const url = `${JUGADORPARTIDO_API_BASE_URL}/jugadoresPartidos`;
        return axios.get(url);
    }

    createJugadorPartido(jugadorPartidoDTO) {
        return axios.post(JUGADORPARTIDO_API_BASE_URL, jugadorPartidoDTO);
    }

    getJugadorPartidoById(jugadorPartidoId) {
        const url = `${JUGADORPARTIDO_API_BASE_URL}/${jugadorPartidoId}`;
        return axios.get(url);
    }

    updateJugadorPartido(jugadorPartidoId, jugadorPartidoDTO) {
        const url = `${JUGADORPARTIDO_API_BASE_URL}/${jugadorPartidoId}`;
        return axios.put(url, jugadorPartidoDTO);
    }

    deleteJugadorPartido(jugadorPartidoId) {
        const url = `${JUGADORPARTIDO_API_BASE_URL}/${jugadorPartidoId}`;
        return axios.delete(url);
    }
}

export default new JugadorPartidoService();